### PR TITLE
Let darktable-bench know where darktable-cli is installed on macOS

### PR DESCRIPTION
**Pull request by victoryforce**

### CODE REVIEW

File: src/tests/benchmark/darktable-bench [hunk 1]
```diff
@@ -66,6 +67,13 @@ def locate_program(program):
             return build+program
          if VERBOSE:
             print(f'  did not find {program} in {build}')
+      if platform.system() == 'Darwin':
+         macpath = '/Applications/darktable.app/Contents/MacOS/'
+         program = 'darktable-cli'
```
Comment (ralfbrown) [inline src/tests/benchmark/darktable-bench:72]: The user could have specified a different executable name than 'darktable-cli'.  This forces the name regardless of commandline args.

**Comment by victoryforce** [comment]
Unless we specify program location explicitly (be it relative or absolute path) darktable-bench tries to find darktable-cli executable by checking the standard locations in the following order:

- in the build directory under the ___source tree___ (assuming we are in the directory of darktable-bench in this tree)
- same thing, but assuming we are in the "sibling" directory relative to the ___source tree___ (well, why not, but it's not obvious to the user unless they've looked at the source code <--- this is a note to self that adding some additional info in the help text would be helpful)

and, if darktable-cli is not found in these directories (such as when darktable just hasn't been compiled yet), then we
- try to locate ___installed___ program

This is how the program behaves on Linux. So, from the user's point of view: _benchmark tries to find the program built in the source tree and if not found, tries to find installed one_.

But on macOS, darktable-bench currently won't find the installed program. That is, so far it has a different behavior on this OS. Why can not find? Because installing an app on macOS from app bundle places it in its own subtree and does not add the executable file to the PATH.

So my PR actually makes darktable-bench behave on macOS consistent with how it behaves on Linux.

> If the PATH variable is properly set I suppose it works, no?

No. See above.

> Also, this could potentially breaks some dev installation having multiple darktable.app installed, no?

Likewise, on Linux I can install multiple darktable versions, but only one installation would be found, the one located in the PATH. To select others, you just need to specify them explicitly. Nothing breaks, really.

Therefore, summing up:

> Do we really want to add a specific OS path hard coded?

We really want to maintain the same behavior regardless of OS: try find built app, if not found try to find the one installed by the official package. Yes, to support this behavior on macOS we have to specify a hard-coded path.

File: src/tests/benchmark/darktable-bench (hunk 1 above)
Comment (victoryforce) [reply src/tests/benchmark/darktable-bench:72]: Yes you're right. Thanks. I will fix it.

**Comment by MStraeten** [comment]
but that doesn't hep if the app package isn't run from Applications but from a mounted package or an arbitrary other place. So if you want to hardcode something then you need to do it relative. 
Especially to run benchmark tests with new an older builds - you won't copy them to Application folder. And also for doing benchmark test with custom builds  you don't want to copy that to your App directory where you're keeping your production build.
So hardcoding an arbitrary edge case doesn't make sense.

**Comment by victoryforce** [comment]
> but that doesn't hep if the app package isn't run from Applications but from a mounted package or an arbitrary other place.

Yes, of course, if the darktable-bench user wants to run darktable-cli from an arbitrary location, _then that location must be specified as a parameter_. How could it be otherwise? No amount of darktable-bench improvement will be able to guess in which arbitrary locations a user may place different builds of darktable. We clearly know only the location of the installed official app bundle. All that is wanted is for the benchmark to be able to find this location by itself without the need for the user to enter this location as a parameter.

I have to repeat what I said above: this PR actually makes darktable-bench behave on macOS consistent with how it behaves on Linux. That's all.

**Comment by MStraeten** [comment]
Theres no value in handling ist like linux.
As long as ist isnt a relative path it‘s just a solution for an arbitrary edge case that will give misleading results in other cases

**Comment by TurboGit** [comment]
Let's close this at this stage.
